src/components/Modal/ModalHeader.tsx

```diff
@@ -0,0 +1,85 @@
+import React from 'react'
+import { Theme, useStyles } from '../../styles'
+import { Heading } from '../Heading'
+import { IconImage } from '../Icon'
+import { Icon, IconColor } from '../Icon/Icon'
+import { HFlow } from '../HFlow'
+import { VFlow } from '../VFlow'
+import { ModalCloseButton } from './ModalCloseButton'
+
+export type HeaderIconObject = {
+  icon: IconImage
+  fill?: IconColor
+  stroke?: IconColor
+}
+
+export type HeaderIconType = IconImage | HeaderIconObject
+
+export interface ModalHeaderProps {
+  title: string
+  subtitle?: string
+  icon?: HeaderIconType
+  background?: string
+  hasCloseIcon?: boolean
+  hasBorder?: boolean
+  onCloseButtonClick?: () => void
+}
+
+export const ModalHeader = (props: ModalHeaderProps) => {
+  const { title, subtitle, icon, background, hasBorder = true, hasCloseIcon = true, onCloseButtonClick } = props
+
+  const { classes } = useStyles(createStyles, background, hasBorder)
+
+  const renderIcon = () => {
```

Comment:
This is a component. You should put it outside the ModalHeader and pass the `icon` as a prop.